### DISCUSSION TITLE
Use of const() leads to wrong result / (another) SyntaxError

### DISCUSSION THREAD

**Discussion by JoGeDuBo** [micropython/micropython#11195] [General]
The comments in the code snippets below explain the problems. 

Tested at RaspberryPi Pico with rp2-pico-20220618-v1.19.1.uf2, 
and also with rp2-pico-20230331-unstable-v1.19.1-995-g0a3600a9a.uf2

Example with syntax error (reason for my investigation):
```python
from micropython import const
def f1():
    C = const(0) # expected to be local to f1

def f2():
    C = 1   # SyntaxError: can't assign to expression
```
Example with (only, but even worse) a wrong result
```python
from micropython import const
A = 1

def f():
    A = const(0) # expected to be local to f

print(f"{A=}")  # result: A=0; expected: A=1
```
As long as it behaves like this, there should be a warning in the documentation, 
that const() always (even if it's used locally) creates/replaces a __global__ name.

Wouldn't it be possible to prevent the use of a pre-existing name in the const-assignment?

**Comment by dlech** [comment]
See https://github.com/micropython/micropython/issues/9867

**Comment by JoGeDuBo** [comment]
Thank you for the link @dlech.
Would be nice to find this comment there of dpgeorge 
"For now, I suggest just making all const variables global, and making their names unique"
in the doc.
